stdlib/list.mli

```diff
@@ -103,9 +103,8 @@ val init : int -> (int -> 'a) -> 'a list
  *)
 
 val append : 'a list -> 'a list -> 'a list
-(** Concatenate two lists. Same function as the infix operator [@].
-   Not tail-recursive (length of the first argument). The [@]
-   operator is not tail-recursive either.
+(** [append l0 l1] appends [l1] to [l0].
+     Same function as the infix operator [@].
```

Comment:
Although I appreciate that 'being tail recursive' is the default in the documentation, this function has been blowing stacks since 1996.  Would it be worth putting in a short note along the lines that 'This function has been tail recursive since OCaml 5.1' onwards to make it explicit when the improvement happened?

Reply:
Good idea, added in [e238433](https://github.com/ocaml/ocaml/pull/11859/commits/e23843308500cf460e77e46d437d5876dd4e4926).